File: api/internal/submission/submission.go
package submission

type MediaType string

const (
	MediaTypeMovie = "movie"
	MediaTypeTV    = "tv"
)

type Severity int

const (
	SeverityMild     Severity = 1
	SeverityModerate Severity = 2
	SeveritySevere   Severity = 3
)

type Submission struct {
	ID       int      `json:"id"`
	MediaID  int      `json:"mediaID"`
	Severity Severity `json:"severity"`
	Start    float64  `json:"start"`
	End      float64  `json:"end"`
	Season   *int     `json:"season,omitempty"`
	Episode  *int     `json:"episode,omitempty"`
	IsReport bool     `json:"isReport"`
}

type AddSubmissionRequest struct {
	Severity Severity `json:"severity"`
	Start    float64  `json:"start"`
	End      float64  `json:"end"`
	Season   *int     `json:"season,omitempty"`
	Episode  *int     `json:"episode,omitempty"`
	IsReport bool     `json:"isReport"`
}

File: extension/src/submission.ts
export type Submission = {
  id: number;
  mediaID: number;
  severity: SubmissionSeverity;
  start: number;
  end: number;
  season?: number;
  episode?: number;
  isReport: boolean;
};

export type SubmissionSeverity = 1 | 2 | 3;
/*
  Severity 1 - Revealing clothing (e.g bikini, lingerie, etc.)
  Severity 2 - Sexually suggestive content (e.g making out, kissing, etc.)
  Severity 3 - Nudity
*/

export type MediaType = "movie" | "tv";
